clients/src/main/java/org/apache/kafka/common/metrics/stats/SampledStat.java

```diff
@@ -110,40 +111,43 @@ public String toString() {
     protected void purgeObsoleteSamples(MetricConfig config, long now) {
         long expireAge = config.samples() * config.timeWindowMs();
         for (Sample sample : samples) {
-            if (now - sample.lastWindowMs >= expireAge)
+            if (now - sample.lastEventMs >= expireAge)
```

Comment:
Could we add a comment that we don't purge overlapping samples?

Reply:
Applied